src/messages/src_authority.rs

```diff
@@ -62,12 +62,22 @@ impl SrcAuthority {
         matches!(self, Self::Section { .. })
     }
 
-    pub(crate) fn as_node(&self) -> Result<(XorName, u8)> {
+    pub(crate) fn to_node_name(&self) -> Result<XorName> {
         match self {
+            Self::Node { public_key, .. } => Ok(name(public_key)),
+            Self::Section { .. } => Err(Error::BadLocation),
+        }
+    }
+
+    pub(crate) fn to_node_peer(&self, addr: Option<SocketAddr>) -> Result<Peer> {
+        match self {
+            Self::Section { .. } => Err(Error::BadLocation),
             Self::Node {
                 public_key, age, ..
-            } => Ok((name(public_key), *age)),
-            Self::Section { .. } => Err(Error::BadLocation),
+            } => {
+                let addr = addr.ok_or(Error::InvalidSource)?;
```

Comment:
seems the input addr has to be a solid value?
why not just having the passed in parameter as a non-option, so that the error case can be handled immediately at the calling place ?

Reply:
This code existed before this PR. I think it's there just for convenience but it's a bit confusing. Will change it.